subworkflows/local/arg.nf

```diff
@@ -0,0 +1,80 @@
+/*
+    Run ARG screening tools
+*/
+
+include { FARGENE                 } from '../../modules/nf-core/modules/fargene/main'
+include { DEEPARG_DOWNLOADDATA    } from '../../modules/nf-core/modules/deeparg/downloaddata/main'
+include { DEEPARG_PREDICT         } from '../../modules/nf-core/modules/deeparg/predict/main'
+
+include { HAMRONIZATION_DEEPARG   } from '../../modules/nf-core/modules/hamronization/deeparg/main'
+include { HAMRONIZATION_SUMMARIZE } from '../../modules/nf-core/modules/hamronization/summarize/main'
+
+workflow ARG {
+    take:
+    contigs // file: /path/to/samplesheet.csv
+    annotations // output from prokka
+
+    main:
+    ch_versions = Channel.empty()
+    ch_mqc      = Channel.empty()
+
+     // Prepare HAMRONIZATION reporting channel
+    ch_input_to_hamronization_summarize = Channel.empty()
+
+    // fARGene run
+    if ( !params.skip_arg_fargene ) {
+        FARGENE ( contigs, params.fargene_hmm_model )
+        ch_versions = ch_versions.mix(FARGENE.out.versions)
+    }
+
+    // DeepARG prepare download
+    if ( !params.skip_arg_deeparg && params.deeparg_data ) {
+        Channel
+            .fromPath( params.deeparg_data )
+            .set { ch_deeparg_db }
+    } else if ( !params.skip_arg_deeparg && !params.deeparg_data ) {
+        DEEPARG_DOWNLOADDATA( )
+        DEEPARG_DOWNLOADDATA.out.db.set { ch_deeparg_db }
+    }
+
+    // DeepARG run
+
+    if ( !params.skip_arg_deeparg ) {
+
+        annotations
+                .map {
+                    it ->
+                        def meta  = it[0]
+                        def anno  = it[1]
+                        def model = params.deeparg_model
+
+                    [ meta, anno, model ]
+                }
+                .set { ch_input_for_deeparg }
+
+        DEEPARG_PREDICT ( ch_input_for_deeparg, ch_deeparg_db )
```

Comment:
Looks like `ch_input_for_deeparg` has many elements but `ch_deeparg_db` has only one element. That would result in `DEEPARG_PREDICT` only being executed once and then `ch_deeparg_db` is exhausted. If that's the case, we can turn it into a value channel that is never exhausted by simply calling `.first()` on it.

Reply:
My understanding was that DSL2 implicitly allows forking in teh background of channels, so you can always reuse

https://www.nextflow.io/docs/latest/dsl2.html#channel-forking

This seems to be the case in the test run, where we download database once, but run it with samples twice:

```
[51/fcf165] process > NFCORE_FUNCSCAN:FUNCSCAN:ARG:DEEPARG_DOWNLOADDATA                          [100%] 1 of 1 ✔
[a1/d900df] process > NFCORE_FUNCSCAN:FUNCSCAN:ARG:DEEPARG_PREDICT (sample_2)                    [100%] 2 of 2 ✔
[b8/c4b2d9] process > NFCORE_FUNCSCAN:FUNCSCAN:ARG:HAMRONIZATION_DEEPARG (sample_2)              [100%] 4 of 4 ✔
```

Reply:
We talked about this on Slack. This works with the download because

> A value channel is implicitly created by a process when an input specifies a simple value in the from clause. Moreover, a value channel is also implicitly created as output for a process whose inputs are only value channels.

https://www.nextflow.io/docs/latest/channel.html#value-channel